src/modules/import/domain/import-event-handler.ts

```diff
@@ -0,0 +1,162 @@
+export class ImportEventHandler {
+    public selectedOrganisationId!: string;
+
+    public selectedRolleId!: string;
+
+    public constructor(
+        private readonly organisationRepository: OrganisationRepository,
+        private readonly importDataRepository: ImportDataRepository,
+        private readonly personenkontextCreationService: PersonenkontextCreationService,
+        private readonly importVorgangRepository: ImportVorgangRepository,
+        private readonly importPasswordEncryptor: ImportPasswordEncryptor,
+        private readonly logger: ClassLogger,
+    ) {}
+
+    @EventHandler(ImportExecutedEvent)
+    public async handleExecuteImport(event: ImportExecutedEvent): Promise<void> {
+        this.selectedOrganisationId = event.organisationId;
+        this.selectedRolleId = event.rolleId;
+
+        //Optimierung: private methode gibt eine map zurück
+        const klassenByIDandName: OrganisationByIdAndName[] = [];
+        const klassen: Organisation<true>[] = await this.organisationRepository.findChildOrgasForIds([
+            this.selectedOrganisationId,
+        ]);
+        klassen.forEach((value: Organisation<true>) => {
+            if (value.typ === OrganisationsTyp.KLASSE) {
+                klassenByIDandName.push({
+                    id: value.id,
+                    name: value.name,
+                });
+            }
+        });
+        // Get all import data items with importvorgangId
+        const importDataItemsWithLoginInfo: ImportDataItem<true>[] = [];
+        //Optimierung: für das folgeTicket mit z.B. 800 Lehrer , muss der thread so manipuliert werden (sobald ein Resultat da ist, wird der nächste request abgeschickt)
+        //Optimierung: Process 10 dataItems at time for createPersonWithPersonenkontexte
+        // const offset: number = 0;
+        // const limit: number = 10;
+
+        const importvorgangId: string = event.importVorgangId;
+        const importVorgang: Option<ImportVorgang<true>> = await this.importVorgangRepository.findById(importvorgangId);
+        if (!importVorgang) {
+            //TODO: Log and return
+            throw new EntityNotFoundError('ImportVorgang', importvorgangId);
+        }
+
+        const [importDataItems, total]: Counted<ImportDataItem<true>> =
+            await this.importDataRepository.findByImportVorgangId(importvorgangId);
+        if (total === 0) {
+            //TODO: Log and return
+            throw new EntityNotFoundError('ImportDataItem', importvorgangId);
+        }
+
+        //create Person With PKs
+        //We must create every peron individually otherwise it cannot assign the correct username when we have multiple users with the same name
+        const savedPersonenWithPersonenkontext: (DomainError | PersonPersonenkontext)[] = [];
+        /* eslint-disable no-await-in-loop */
+        for (const importDataItem of importDataItems) {
+            const klasse: OrganisationByIdAndName | undefined = klassenByIDandName.find(
```

Comment:
Statt der for loop könnte man ein async .map verwenden und dann intern die Logik in eine Funktion auslagern um so die ganzen awaits in der for loop zu vermeiden.

Reply:
Das kann nicht funktionieren, ich hatte das schonmal probiert. So kann man nicht eine CSV Datei importieren, die mindestens 2 Nutzer mit gleichen Namen, weil das führt dazu dass die beiden Users den gleichen Username generiert bekommen aber der zweite Usernaname kann nicht in KC angelegt werden weil den gibt es theoretisch schon.